lang/src/org/partiql/lang/eval/Thunk.kt

```diff
@@ -133,7 +133,7 @@ internal abstract class ThunkFactory(
         // When this check is enabled we throw an exception the [MetaContainer] does not have a
         // [StaticTypeMeta].  This indicates a bug or unimplemented support for an AST node in
         // [StaticTypeInferenceVisitorTransform].
-        val staticType = metas.staticType?.type ?: error("Metas collection does not have a StaticTypeMeta")
+        val staticType = (metas["\$static_type"]as StaticTypeMeta?)?.type ?: error("Metas collection does not have a StaticTypeMeta")
```

Comment:
Can you modify the `.staticType` value for `StaticTypeMeta` rather than passing in the string: https://github.com/partiql/partiql-lang-kotlin/blob/03b30653480d1d315dd321e5305183d93e2efd9e/lang/src/org/partiql/lang/ast/StaticTypeMeta.kt#L82

Reply:
Will make the change in the next commit.